### ISSUE TITLE
--allow-net ignores ports

### ISSUE THREAD

**Issue by teleclimber** [denoland/deno#21220]
Consider this script:

```
console.log(await fetch("http://localhost:3003"));
```

If I run it with:
```
deno run --allow-net=localhost:80 intex.ts
```

...it runs without complaining and logs a response from the server at 3003.

If I allow a non standard port, like 5000, it prompts for permission for :3003 specifically.

This appears to be a regression of https://github.com/denoland/deno/issues/6531

I tested on Deno 1.35.1 and 1.38.1 with the same result.

**Comment by irbull** [comment]
This does appear to be a regression, somewhere between Deno 1.8 and 1.9.

With the following:
```
console.log(await fetch("https://www.google.com"));
```

If I only allow access to port 80 (`deno run --allow-net=www.google.com:80 test.ts`).

With 1.8:
```
Check file:///Users/irbull/git/deno/lsp-test/test.ts
error: Uncaught PermissionDenied: network access to "www.google.com", run again with the --allow-net flag
console.log(await fetch("https://www.google.com"));
```

With 1.9, deno will fetch from 443 without a problem.

I don't know if this was a design decision back then.

**Comment by benatkin** [comment]
A built-in reverse proxy of sorts would be needed, but I would like to be able to listen only on a hostname, so I could do:

```
deno run --allow-net=mylilapp.localhost
```

...and then if I listened on that, all incoming HTTP requests with a different Host: field would be rejected

It would basically mean that Deno's HTTP serve APIs would be repurposed from just speaking HTTP to either speaking HTTP to the outside world or speaking it to a reverse proxy, and it would have to reject non-HTTP (or non-host-providing APIs)

That would be some extra complexity but it would make it easier to make a web app that doesn't use other apps

**Comment by irbull** [comment]
It seems a bit more subtle than this. It seems only related to port 80, that is, if a host and specifically port `80` are specified (as in `example.com:80`) then all traffic to that host are allowed. If a port other than 80 is specified, then the permissions properly restrict access. I've created a PR that adds a test for this. I'll see if I can find out where this issue is coming from.

**Comment by teleclimber** [comment]
I concur. I would have thought `:443` would behave like `:80`, but somehow 443 is handled correctly while 80 is not. (In my experiments on 1.38.1 at least).

**Comment by teleclimber** [comment]
Oh cool I can actually close this! Verified as fixed in 1.39.4. Thanks @irbull !
